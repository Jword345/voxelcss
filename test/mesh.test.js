'use strict';

const ColorFace = require('../lib/ColorFace');
const Mesh      = require('../lib/Mesh');

describe('Mesh', function() {
    let mesh;
    beforeEach(() => {
        mesh = new Mesh();
    });
    afterEach(() => {
        mesh = undefined;
    });
    it('can be serialized', () => {
        expect(mesh.serialize()).toMatchSnapshot();
    });
    it('can be loaded from serialized string', () => {
        let testMesh = new Mesh(new ColorFace('#333'));
        let meshString = testMesh.serialize();
        mesh = Mesh.loadFromSerial(meshString);
        expect(mesh.serialize()).toEqual(meshString);
    });
    it('can get and set faces', () => {
        expect(mesh.getFaces()).toMatchSnapshot();
        expect(mesh.getFront()).toMatchSnapshot();
        expect(mesh.getBack()).toMatchSnapshot();
        expect(mesh.getLeft()).toMatchSnapshot();
        expect(mesh.getRight()).toMatchSnapshot();
        expect(mesh.getTop()).toMatchSnapshot();
        expect(mesh.getBottom()).toMatchSnapshot();
    });
    it('can do nothing when invalid face data is passed', function() {
        let invalidFaceData = 'Not a valid face';
        expect(mesh.setFront(invalidFaceData)).toBeUndefined();
        expect(mesh.setBack(invalidFaceData)).toBeUndefined();
        expect(mesh.setLeft(invalidFaceData)).toBeUndefined();
        expect(mesh.setRight(invalidFaceData)).toBeUndefined();
        expect(mesh.setTop(invalidFaceData)).toBeUndefined();
        expect(mesh.setBottom(invalidFaceData)).toBeUndefined();
    });
    it('can trigger change event when a face is changed', function() {
        let faces = mesh.getFaces();
        let front = mesh.getFront();
        expect(mesh.setFaces()).toMatchSnapshot();
        mesh.setFaces(faces);
        mesh.setFront(front);
        expect(mesh.getFront()).toEqual(front);
        front.triggerEvent('change');
    });
});
