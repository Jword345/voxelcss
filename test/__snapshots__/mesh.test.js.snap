// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mesh can be serialized 1`] = `"{\\"front\\":\\"image(data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7)\\",\\"back\\":\\"image(data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7)\\",\\"left\\":\\"image(data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7)\\",\\"right\\":\\"image(data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7)\\",\\"top\\":\\"image(data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7)\\",\\"bottom\\":\\"image(data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7)\\"}"`;

exports[`Mesh can get and set faces 1`] = `
Object {
  "back": ImageFace {
    "addEventListener": [Function],
    "clone": [Function],
    "getSource": [Function],
    "removeEventListener": [Function],
    "serialize": [Function],
    "setSource": [Function],
    "triggerEvent": [Function],
  },
  "bottom": ImageFace {
    "addEventListener": [Function],
    "clone": [Function],
    "getSource": [Function],
    "removeEventListener": [Function],
    "serialize": [Function],
    "setSource": [Function],
    "triggerEvent": [Function],
  },
  "front": ImageFace {
    "addEventListener": [Function],
    "clone": [Function],
    "getSource": [Function],
    "removeEventListener": [Function],
    "serialize": [Function],
    "setSource": [Function],
    "triggerEvent": [Function],
  },
  "left": ImageFace {
    "addEventListener": [Function],
    "clone": [Function],
    "getSource": [Function],
    "removeEventListener": [Function],
    "serialize": [Function],
    "setSource": [Function],
    "triggerEvent": [Function],
  },
  "right": ImageFace {
    "addEventListener": [Function],
    "clone": [Function],
    "getSource": [Function],
    "removeEventListener": [Function],
    "serialize": [Function],
    "setSource": [Function],
    "triggerEvent": [Function],
  },
  "top": ImageFace {
    "addEventListener": [Function],
    "clone": [Function],
    "getSource": [Function],
    "removeEventListener": [Function],
    "serialize": [Function],
    "setSource": [Function],
    "triggerEvent": [Function],
  },
}
`;

exports[`Mesh can get and set faces 2`] = `
ImageFace {
  "addEventListener": [Function],
  "clone": [Function],
  "getSource": [Function],
  "removeEventListener": [Function],
  "serialize": [Function],
  "setSource": [Function],
  "triggerEvent": [Function],
}
`;

exports[`Mesh can get and set faces 3`] = `
ImageFace {
  "addEventListener": [Function],
  "clone": [Function],
  "getSource": [Function],
  "removeEventListener": [Function],
  "serialize": [Function],
  "setSource": [Function],
  "triggerEvent": [Function],
}
`;

exports[`Mesh can get and set faces 4`] = `
ImageFace {
  "addEventListener": [Function],
  "clone": [Function],
  "getSource": [Function],
  "removeEventListener": [Function],
  "serialize": [Function],
  "setSource": [Function],
  "triggerEvent": [Function],
}
`;

exports[`Mesh can get and set faces 5`] = `
ImageFace {
  "addEventListener": [Function],
  "clone": [Function],
  "getSource": [Function],
  "removeEventListener": [Function],
  "serialize": [Function],
  "setSource": [Function],
  "triggerEvent": [Function],
}
`;

exports[`Mesh can get and set faces 6`] = `
ImageFace {
  "addEventListener": [Function],
  "clone": [Function],
  "getSource": [Function],
  "removeEventListener": [Function],
  "serialize": [Function],
  "setSource": [Function],
  "triggerEvent": [Function],
}
`;

exports[`Mesh can get and set faces 7`] = `
ImageFace {
  "addEventListener": [Function],
  "clone": [Function],
  "getSource": [Function],
  "removeEventListener": [Function],
  "serialize": [Function],
  "setSource": [Function],
  "triggerEvent": [Function],
}
`;

exports[`Mesh can trigger change event when a face is changed 1`] = `
Object {
  "back": ImageFace {
    "addEventListener": [Function],
    "clone": [Function],
    "getSource": [Function],
    "removeEventListener": [Function],
    "serialize": [Function],
    "setSource": [Function],
    "triggerEvent": [Function],
  },
  "bottom": ImageFace {
    "addEventListener": [Function],
    "clone": [Function],
    "getSource": [Function],
    "removeEventListener": [Function],
    "serialize": [Function],
    "setSource": [Function],
    "triggerEvent": [Function],
  },
  "front": ImageFace {
    "addEventListener": [Function],
    "clone": [Function],
    "getSource": [Function],
    "removeEventListener": [Function],
    "serialize": [Function],
    "setSource": [Function],
    "triggerEvent": [Function],
  },
  "left": ImageFace {
    "addEventListener": [Function],
    "clone": [Function],
    "getSource": [Function],
    "removeEventListener": [Function],
    "serialize": [Function],
    "setSource": [Function],
    "triggerEvent": [Function],
  },
  "right": ImageFace {
    "addEventListener": [Function],
    "clone": [Function],
    "getSource": [Function],
    "removeEventListener": [Function],
    "serialize": [Function],
    "setSource": [Function],
    "triggerEvent": [Function],
  },
  "top": ImageFace {
    "addEventListener": [Function],
    "clone": [Function],
    "getSource": [Function],
    "removeEventListener": [Function],
    "serialize": [Function],
    "setSource": [Function],
    "triggerEvent": [Function],
  },
}
`;
